editor/components/CrashDiagramCard.tsx

```diff
@@ -71,52 +95,94 @@ const ZoomResetControls = ({
         <Button
           size="sm"
           variant="outline-primary"
-          onClick={() => zoomIn(0.25)}
+          onClick={() => {
+            const newScale = instance.transformState.scale + 0.25;
+            zoomIn(0.25);
+            setValue("scale", newScale, {
+              shouldDirty: true,
+            });
+          }}
           title="Zoom In"
         >
-          <FaMagnifyingGlassPlus />
+          <AlignedLabel>
+            <FaMagnifyingGlassPlus />
+          </AlignedLabel>
         </Button>
         <Button
           size="sm"
           variant="outline-primary"
-          onClick={() => zoomOut(0.25)}
+          onClick={() => {
+            const newScale = instance.transformState.scale - 0.25;
+            zoomOut(0.25);
+            setValue("scale", newScale, {
+              shouldDirty: true,
+            });
+          }}
           title="Zoom Out"
         >
-          <FaMagnifyingGlassMinus />
+          <AlignedLabel>
+            <FaMagnifyingGlassMinus />
+          </AlignedLabel>
+        </Button>
+        <Button
+          size="sm"
+          variant="outline-primary"
+          onClick={handleReset}
+          title="Reset"
+        >
+          <AlignedLabel>
+            <SlActionUndo className="me-2" />
+            Reset
+          </AlignedLabel>
         </Button>
       </ButtonGroup>
-      <Button
-        size="sm"
-        variant="outline-primary"
-        onClick={handleReset}
-        title="Reset"
-      >
-        <SlActionUndo className="me-2" />
-        Reset
-      </Button>
+      <PermissionsRequired allowedRoles={allowedUserSaveDiagramRoles}>
+        <AlignedLabel>
+          <Button
+            size="sm"
+            variant={"primary"}
+            onClick={handleSubmit(onSave)}
+            title={"save"}
+            disabled={!isDirty}
+          >
+            <AlignedLabel>
+              {(isDirty || (!isDirty && !isSaved)) && (
+                <>
+                  <FaFloppyDisk className="me-2" />
+                  Save
+                </>
+              )}
+              {isSaved && !isDirty && (
+                <>
+                  <FaCheckCircle className="me-2" />
+                  Saved
+                </>
+              )}
+            </AlignedLabel>
```

Comment:
I know y'all have been iterating on this a lot already, but I'm wondering if there could be a third state that indicates saving will overwrite existing data? Not sure if that would make it more confusing, though. 🤔